### PR TITLE
Multiple UI tunes

### LINKED ISSUE #3965: lazy load feed images
The background image loading is super resource intensive for large feeds. Things like this should work:

https://github.com/Pentiado/angular-lazy-img

### PR DESCRIPTION
**Pull request by fffw**
Fix #3965 and partially https://github.com/getlantern/lantern/issues/3959

### CODE REVIEW

File: src/github.com/getlantern/lantern-ui/app/_css/news.css [hunk 1]
```diff
@@ -36,73 +36,78 @@
 }
 
 #news .tab-content {
+  overflow: hidden;
+}
+
+#news .feeds-container {
```
Comment (fffw) [inline src/github.com/getlantern/lantern-ui/app/_css/news.css]: Added another layer to let infinite scroll work

File: src/github.com/getlantern/lantern-ui/app/_css/news.css [hunk 2]
```diff
@@ -118,20 +123,14 @@
...
-}*/
-
 #news .feed-thumbnail {
   float: left;
   width: 145px;
   height: 88px;
   margin: 17px 21px;
+}
+
+#news .feed-thumbnail img {
```
Comment (fffw) [inline src/github.com/getlantern/lantern-ui/app/_css/news.css]: Wrap image in a fixed width/height div, so image ratio can be kept.

File: src/github.com/getlantern/lantern-ui/app/_css/news.css [hunk 3]
```diff
@@ -151,11 +150,36 @@
 
 #news h4 {
   font-weight: 700;
-  line-height: 16px;
+  line-height: 18px;
 }
 
+/* http://hackingui.com/front-end/a-pure-css-solution-for-multiline-text-truncation/ */
```
Comment (fffw) [inline src/github.com/getlantern/lantern-ui/app/_css/news.css]: This whole bunch of code is to add `...` if text is truncated.

File: src/github.com/getlantern/lantern-ui/app/angular-feed/styles/spinner.css [hunk 4]
```diff
@@ -45,13 +45,14 @@
 }
 
 .spinner {
-    position:relative;
```
Comment (fffw) [inline src/github.com/getlantern/lantern-ui/app/angular-feed/styles/spinner.css]: Center the spinner

File: src/github.com/getlantern/lantern-ui/app/js/controllers.js [hunk 5]
```diff
@@ -186,16 +186,44 @@ app.controller('NewsfeedCtrl', ['$scope', '$rootScope', '$translate', function($
...
+    $scope.tabActive[title] = false;
+  };
+  $scope.tabSelected = function (title) {
+    return $scope.tabActive[title] === true;
+  };
 }]);
 
 app.controller('FeedCtrl', ['$scope', 'gaMgr', function($scope, gaMgr) {
+  var copiedFeedEntries = [];
+  angular.copy($scope.feedEntries, copiedFeedEntries);
```
Comment (fffw) [inline src/github.com/getlantern/lantern-ui/app/js/controllers.js]: Make a deep copy, so `splice` doesn't modify the original object.

File: src/github.com/getlantern/lantern-ui/app/js/controllers.js [hunk 6]
```diff
@@ -186,16 +186,44 @@ app.controller('NewsfeedCtrl', ['$scope', '$rootScope', '$translate', function($
...
+  $scope.entries = [];
+  $scope.containerId = function($index) {
+    return "#feeds-container-" + $index;
+  };
+  $scope.addMoreItems = function() {
+    if ($scope.tabSelected($scope.feedsTitle)) {
```
Comment (fffw) [inline src/github.com/getlantern/lantern-ui/app/js/controllers.js]: Add items only if tab is active, the key to make infinite scroll work for multiple tabs.

**Comment by myleshorton** [comment]
For some reason it's not loading a lot of the images for me now @fffw. Also the thumbnails are all now bleeding off each entry:

image

**Comment by fffw** [comment]
Oh I unintentionally changed the behaviour of thumbnail. It should look fine now @myleshorton

**Comment by myleshorton** [comment]
Awesome @fffw it's looking great to me now! Good stuff on the aspect ratio too.

**Comment by myleshorton** [comment]
I'll just look at your changes in the code a little more carefully, and then I'll just merge and build.

File: src/github.com/getlantern/lantern-ui/app/_css/news.css (hunk 2 above)
Comment (myleshorton) [inline src/github.com/getlantern/lantern-ui/app/_css/news.css]: Nice @fffw.

File: src/github.com/getlantern/lantern-ui/app/js/controllers.js (hunk 6 above)
Comment (myleshorton) [inline src/github.com/getlantern/lantern-ui/app/js/controllers.js]: nice!!!

**Comment by myleshorton** [comment]
Looking great @fffw!

**Pull request merged**